iina/LogWindowController.swift

```diff
@@ -40,28 +23,59 @@ class LogWindowController: NSWindowController, NSTableViewDelegate, NSTableViewD
   @IBOutlet weak var subsystemPopUpButton: NSPopUpButton!
   @IBOutlet weak var levelPopUpButton: NSPopUpButton!
 
-  @objc dynamic var logs: [Log] = []
+  @objc dynamic var logs: [Logger.Log] {
+    Logger.logs
+  }
   @objc dynamic var predicate = NSPredicate(value: true)
 
   override func windowDidLoad() {
     super.windowDidLoad()
 
     logTableView.sizeLastColumnToFit()
     let tableViewMenu = NSMenu()
-    tableViewMenu.insertItem(withTitle: "Copy", action: #selector(menuCopy), keyEquivalent: "", at: 0)
+    tableViewMenu.addItem(withTitle: "Copy", action: #selector(menuCopy), keyEquivalent: "")
     logTableView.menu = tableViewMenu
 
     levelPopUpButton.menu?.items.forEach {
-      $0.image = NSImage.circle(withColor: colorMap[$0.tag]!)
+      $0.image = LogWindowController.indicatorIcon(withColor: colorMap[$0.tag]!)
     }
+    levelPopUpButton.selectItem(withTag: Logger.Level.preferred.rawValue)
     subsystemPopUpButton.menu!.delegate = self
-    subsystemPopUpButton.selectItem(withTag: Preference.integer(for: .logLevel))
+  }
+
+  fileprivate static func indicatorIcon(withColor color: NSColor) -> NSImage {
+    if let cached = circleDict[color] {
+      return cached
+    }
+    let image = NSImage(size: NSMakeSize(kIconSize, kIconSize), flipped: false) { rect in
+      let inset = NSInsetRect(rect, kBorderWidth / 2 + rect.size.width * 0.25, kBorderWidth / 2 + rect.size.height * 0.25)
+      let path = NSBezierPath.init(ovalIn: inset)
+      path.lineWidth = kBorderWidth
+
+      let fractionOfBlendedColor: CGFloat
+      if #available(macOS 10.14, *) {
+        fractionOfBlendedColor = (NSApp.appearance?.isDark ?? false) ? 0.15 : 0.3
+      } else {
+        fractionOfBlendedColor = 0.15
+      };
+      let borderColor = color.blended(withFraction: fractionOfBlendedColor, of: .controlTextColor)
+
+      borderColor?.setStroke()
+      path.stroke()
+
+      color.setFill()
+      path.fill()
+
+      return true
+    }
+    circleDict[color] = image
+    return image
   }
 
   // NSMenuDelegate
```

Comment:
I would expect this comment to be:
```swift
// MARK: - NSMenuDelegate
```
In order to support using the Xcode jump bar as discussed in: [Editing source files in Xcode](https://developer.apple.com/documentation/xcode/creating-organizing-and-editing-source-files)

Reply:
I confirmed this has been addressed.